packages/webdriverio/src/utils/index.ts

```diff
@@ -427,6 +432,13 @@ export async function findDeepElements(
     return deepElementResult as ElementReference[]
 }
 
+/**
+* Temporary patch for https://github.com/mozilla/geckodriver/issues/2223
+*/
+function returnUniqueNodes(nodes: ExtendedElementReference[]): ExtendedElementReference[] {
+    return Array.from(new Set(nodes.map(node => JSON.stringify(node)))).map(res => JSON.parse(res))
```

Comment:
I believe we can filter using the element id's only, this way we don't have to do transform to a JSON string and back to an object and we also don't need to map